formvalidator/src/main/java/dev/poteto/formvalidator/rules/ValidateCurrency.kt

```diff
@@ -0,0 +1,44 @@
+package dev.poteto.formvalidator.rules
+
+import dev.poteto.formvalidator.DataType
+import dev.poteto.formvalidator.contracts.PredefinedValidationRule
+import dev.poteto.formvalidator.contracts.ValidationMessages
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+/**
+ * Class for validating Currency
+ * @param target inherit
+ * @param messages inherit
+ * @param type inherit
+ */
+class ValidateCurrency(
+    target: CharSequence?,
+    messages: ValidationMessages,
+    type: DataType
+) : PredefinedValidationRule(target, messages, type) {
+
+    /**
+     * Check the email is valid or not with regex method
+     */
+    override fun check(): Boolean {
+        val p: Pattern = Pattern.compile("^(?:0|[1-9]\\d{0,2}(?:\\.\\d{3})*),\\d{2}$")
+        val m: Matcher = p.matcher(target.toString())
+        if (!m.matches()){
+            return false
+        }
+        return true
+    }
+
+    /**
+     * @return message when validation fails
+     */
+    override fun getMessage(): String {
+        val p: Pattern = Pattern.compile("^(?:0|[1-9]\\d{0,2}(?:\\.\\d{3})*),\\d{2}$")
+        val m: Matcher = p.matcher(target.toString())
+        if (!m.matches()){
+            return messages.currency()
+        }
+        return ""
```

Comment:
There's no need for rechecking the field again. It's a spaghetti code, better to take of it and just return the error message only. If the validation succeed, it would not change the error message of the EditText instance. See [error callback invocation](https://github.com/rendyananta/android-form-validator/blob/cda41a71dc927e2882702527ceeb4269445cd8ad/formvalidator/src/main/java/dev/poteto/formvalidator/RulesDefinition.kt#L68-L77)

```suggestion
        return messages.currency()
```